hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -582,14 +582,16 @@ def b64decode(s):
 
 try:
     from django.test import TransactionTestCase
-    from hypothesis.extra.django import HypothesisTestCase
 
     def bad_django_TestCase(runner):
         if runner is None:
             return False
-        return isinstance(runner, TransactionTestCase) and not isinstance(
-            runner, HypothesisTestCase
-        )
+        if not isinstance(runner, TransactionTestCase):
+            return False
+
+        from hypothesis.extra.django._impl import HypothesisTestCase
```

Comment:
Can you elaborate on what happened here that prompted this move?

Reply:
Basically the same logic as #1703 - in many editors and tools, there was nothing to distinguish `HypothesisTestCase` as a private API.  Moving into an `_`-prefixed namespace makes that clear, though, and ensures that every visible name in `hypothesis.extra.django` is public.